src/rez/resolved_context.py

```diff
@@ -1296,9 +1296,12 @@ def execute_command(self, args, parent_environ=None, **Popen_args):
         Note:
             This does not alter the current python session.
         """
-        if parent_environ in (None, os.environ):
-            target_environ = {}
```

Comment:
This might have been intentional, though the "in os.environ" seems weird to me.

Reply:
I suspect it maybe made sense prior to this line changing:

https://github.com/AcademySoftwareFoundation/rez/blame/3d0f22489827f8def994cb2e5590d51445e55123/src/rez/rex.py#L632

BUt this code now just sets target_environ to an empty dict if called with `parent_environ = os.environ` - which can't possibly be desirable behaviour (if caller of `execute_command` wants to do that, they can just pass `parent_environ={}`)